src/transformers/models/encoder_decoder/modeling_flax_encoder_decoder.py

```diff
@@ -0,0 +1,853 @@
+
+        return FlaxSeq2SeqLMOutput(
+            logits=decoder_outputs.logits,
+            decoder_hidden_states=decoder_outputs.hidden_states,
+            decoder_attentions=decoder_outputs.attentions,
+            cross_attentions=decoder_outputs.cross_attentions,
+            encoder_last_hidden_state=encoder_outputs.last_hidden_state,
+            encoder_hidden_states=encoder_outputs.hidden_states,
+            encoder_attentions=encoder_outputs.attentions,
+        )
+
+
+@add_start_docstrings(ENCODER_DECODER_START_DOCSTRING)
+class FlaxEncoderDecoderModel(FlaxPreTrainedModel):
+    r"""
+    :class:`~transformers.FlaxEncoderDecoderModel` is a generic model class that will be instantiated as a transformer
+    architecture with the module (flax.nn.Module) of one of the base model classes of the library as encoder module and
+    another one as decoder module when created with the :meth`~transformers.FlaxAutoModel.from_pretrained` class method
+    for the encoder and :meth`~transformers.FlaxAutoModelForCausalLM.from_pretrained` class method for the decoder.
+    """
+    config_class = EncoderDecoderConfig
+    base_model_prefix = "encoder_decoder"
+    module_class = FlaxEncoderDecoderModule
+
+    def __init__(
+        self,
+        config: EncoderDecoderConfig,
+        input_shape: Optional[Tuple] = None,
+        seed: int = 0,
+        dtype: jnp.dtype = jnp.float32,
+        **kwargs
+    ):
+        # Q: unlike in `modeling_flax_bart.FlaxBartPreTrainedModel.__init__`, here we use a more general shape
+        # (keep vision models' argument `pixel_values` as a next step).
+        if input_shape is None:
+            input_shape = ((1, 1), (1, 1))
+
+        module = self.module_class(config=config, dtype=dtype, **kwargs)
+        super().__init__(config, module, input_shape=input_shape, seed=seed, dtype=dtype)
+
+        # Copied from modeling.encoder_decoder.EncoderDecoderModel
+        # Q: should we keep this (modified) check - because we don't have `get_output_embeddings` as in PyTorch models.
+        assert not hasattr(
+            self.module._get_encoder_module(), "lm_head"
+        ), "The encoder {} should not have a LM Head. Please use a model without LM Head"
+
+    def init_weights(self, rng: jax.random.PRNGKey, input_shape: Tuple) -> FrozenDict:
+        encoder_input_shape, decoder_input_shape = input_shape
+
+        # init input tensors
+        input_ids = jnp.zeros(encoder_input_shape, dtype="i4")
+        attention_mask = jnp.ones_like(input_ids)
+        decoder_input_ids = jnp.zeros(decoder_input_shape, dtype="i4")
+        decoder_attention_mask = jnp.ones_like(decoder_input_ids)
+
+        batch_size, sequence_length = input_ids.shape
+        position_ids = jnp.broadcast_to(jnp.arange(sequence_length)[None, :], (batch_size, sequence_length))
+
+        decoder_batch_size, decoder_sequence_length = decoder_input_ids.shape
+        # Q: Is it ok to keep this sanity check?
```

Comment:
It would be nice to make it a:

```if decoder_batch_size != batch_size: 
   raise ValueError(....)
```
check instead, but sure it's a good check!